src/components/transport_manager/src/transport_adapter/threaded_socket_connection.cc

```diff
@@ -157,12 +166,16 @@ TransportAdapter::Error ThreadedSocketConnection::SendData(
 TransportAdapter::Error ThreadedSocketConnection::Disconnect() {
   LOG4CXX_AUTO_TRACE(logger_);
   terminate_flag_ = true;
+  if (socket_ != -1) {
+    shutdown(socket_, SHUT_RDWR);
```

Comment:
@AByzhynar I think you are right, I shall investigate this and add shutdown() into Finalize() method too.